File: app/src/main/java/com/example/weatherapp/ArpltnResult.java
package com.example.weatherapp;

import java.util.List;

public class ArpltnResult implements APIResult<PmData> {
    Response response;

    @Override
    public PmData toAppData() throws Exception {
        if (!response.header.resultCode.equals("00")) throw new Exception();
        ArpltnResult.Item item = this.response.body.items.get(0);

        PmData pmData = new PmData();
        pmData.setFields(item, 0);
        pmData.setFields(item, 1);

        return pmData;
    }

    static class Response {
        Header header;
        Body body;
    }

    static class Header {
        String resultCode;
        String resultMsg;
    }

    static class Body {
        int numOfRows;
        int pageNo;
        int totalCount;
        List<Item> items;
    }

    static class Item {
        String pm10Value;
        String pm10Value24;
        String pm25Value;
        String pm25Value24;
        String pm10Flag;
        String pm25Flag;
        String pm10Grade1h;
        String pm10Grade;
        String pm25Grade1h;
        String pm25Grade;
    }
}

File: app/src/main/java/com/example/weatherapp/data/source/remote/alpltn/ArpltnResult.kt
package com.example.weatherapp.data.source.remote.alpltn

class ArpltnResult {
    var response: Response? = null

    class Response {
        var header: Header? = null
        var body: Body? = null
    }

    class Header {
        var resultCode: String? = null
        var resultMsg: String? = null
    }

    class Body {
        var numOfRows = 0
        var pageNo = 0
        var totalCount = 0
        var items: List<Item>? = null
    }

    class Item {
        var pm10Value: String? = null
        var pm10Value24: String? = null
        var pm25Value: String? = null
        var pm25Value24: String? = null
        var pm10Flag: String? = null
        var pm25Flag: String? = null
        var pm10Grade1h: String? = null
        var pm10Grade: String? = null
        var pm25Grade1h: String? = null
        var pm25Grade: String? = null
    }
}